teps/0035-doc-policy-authentication-authorization-rbac.md

```diff
@@ -0,0 +1,635 @@
+stringent offerings, which undoubtedly are even used as the K8s cluster's configured authentication implementation.
+
+### Notes/Constraints/Caveats (optional)
+
+
+### Risks and Mitigations
+
+"Security first" is a mantra often heard amongst Kubernetes cluster administrators.  The fact that some form of 
+security enforcement is not on by default very possibly will come under scrutiny.  
+
+For mitigation on that point, we can use Kubernetes experience around `PodSecurityPolicy` as an example to learn from.
+In particular, the pattern of a new Kubernetes cluster deployment first standing up as "non-secure", and then 
+migrating to more secure usage on a namespace by namespace basis arose early in the ecosystem.  `PodSecurityPolicy` and
+its all or nothing approach did not allow for that pattern and hence it was met with resistance.  The `sig-auth` group
+in charge of `PodSecurityPolicy` has noted that shortcoming as a failing both at various KubeCon's and in the still
+[active KEP](https://github.com/kubernetes/enhancements/issues/5) around `PodSecurityPolicy`.
+
+And certainly use of for example any OPA integrations can be employ from the start whenever a new cluster is created if such an
+approache is tenable.
+
+So for now, as we consider this for Tekton, we are focused on facilitating easy migration to more security.
+
+### User Experience (optional)
+
+The documentation we are looking to provide stems from the motivation to assist the user experience in the
+security/policy/authentication/authorization relam.
+
+### Performance (optional)
+
+Some points of note with OPA Gatekeeper's implementation and performance:
+
+- it injects itself into the Kubernetes cluster as and admission webhook
+- the `match` section of the `spec` in the `Constraint` OPA CRD serves as a filter for which API objects the webhook deals with
+- Gatekeeper employs the Kubernetes controller / watch / informer constructs for providing that data requested in 
+`parameters` section of the `spec` in the `Constraint` OPA CRD.  What that means is the latest versions of those objects
+are obtained in the background, and then when the `ConstraintTemplate` is executed, instead of explicit calls to the
+API server to retrieve those objects, the controller cache is leveraged.
+
+Between the `match` filtering and use of a controller cache, and that general admission webhook overhead is generally accepted
+in the Kubernetes ecosystem as more than tolerable, the only overhead of note will be OPA's Rego processing engine.
+
+
+
+## Design Details
+
+At this time, with documentations and possibly references to policies hosted outside of https://github.com/tektoncd
+there is no coding design within Tekton to declare.
+
+We are building upon the fact that prior K8s design choices that allow for plugging in such solutions (i.e. admission
+webhooks) is the key design point here.
+ 
```

Comment:
do you plan to start with an .md file in the pipelines repo, or somewhere else, or do you think it could make sense to have a separate repo for this? (even if it was a little sparse) i could see the separate repo making sense if you wanted to have any specialized test automation, but maybe just starting with an .md file in an existing repo makes sense for now?

Reply:
Yeah I want to start simple with a .md file in the pipelines repo in the docs folder there.  Users are presumably already used to going there, and that is then the best place to make this subject matter available to them.

I would not be shocked at all that if things evolve in this space and gain traction, we can then evolve to a separate repo.

And I think you point about specialized test automation in spot on ... if we get to that point, a separate repo really becomes the way we should go.